src/04_language.md

```diff
@@ -261,13 +261,53 @@ Beside panics, secure-critical drop should be protected.
 
 ### `Send` and `Sync` traits
 
-The `Send` and `Sync` traits are marker traits that allow implementors
-to specify respectively that a value can be sent (moved) to another thread and
-that a value be shared through a shared reference. They both are unsafe traits,
-that should be used. Moreover, the `unsafe` keyword is necessary for a type
-to implement those traits.
+The `Send` and `Sync` traits (defined in `std::marker` or `core::marker`) are
+marker traits used to ensure the safety of concurrency in Rust. When implemented
+correctly, they allow the Rust compiler to guarantee the absence of data races.
+Their semantics is as follows:
+
+- A type is `Send` if it is safe to send (move) it to another thread.
+- A type is `Sync` if it is safe to share a immutable reference to it with
+  another thread.
+
+Both traits are _unsafe traits_, i.e., the Rust compiler do not verify in any
+way that they are implemented correctly. The danger is real: an incorrect
+implementation may lead to **undefined behavior**.
```

Comment:
To clarify: closures spawned in another thread require that the environment they capture be `Send`, which in turn requires that variables shared across multiple threads be `Sync`; thus **the `Sync` trait plays a key role to prevent multi-threaded data races**.